processor/tailsamplingprocessor/config.go

```diff
@@ -208,6 +213,12 @@ type BooleanAttributeCfg struct {
 	Value bool `mapstructure:"value"`
 }
 
+// OTTLStatementCfg holds the configurable setting to create a OTTL statement filter
+// sampling policy evaluator.
+type OTTLStatementCfg struct {
+	Statements []string `mapstructure:"statements"`
```

Comment:
I strongly recommend you let users decide which ottl.ErrorMode is used with the statements. Checkout transformprocessor or filterprocessor for an example of that configuration.

Reply:
@TylerHelmuth Thanks for the early review! I **love** how our community pays attention to new contributions.

I've awarded how those configs should be decided (mostly comes from user input, like how we've done in transform processor / other internal pkg).

The current code is written like this due to the lack of understanding of OTTL-related packages. I am trying to implement and run it first. This should be well optimized, especially on how OTTL pkg is used, and how the code structure is organized.

Actually, the field you mentioned is an option for config in previous codes, like how we are doing in: `processor/transformprocessor/internal/common/traces.go`
```go
func (pc TraceParserCollection) ParseContextStatements(contextStatements ContextStatements) (consumer.Traces, error) {
                ...
		sStatements := ottlspan.NewStatements(parsedStatements, pc.settings, ottlspan.WithErrorMode(pc.errorMode))
		...
```

I removed them for a quick test. 

I think we will post a correct commit for review later and remove the WIP prefix before ping ya for review. And again your early suggestions are valuable for rookies like me. Please feel free to point out more issues ^_^